src/DateTimePeriod.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of Business.
+ *
+ * (c) Florian Voutzinos <florian@voutzinos.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Business;
+
+/**
+ * DateTime period implementation.
+ *
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+final class DateTimePeriod implements \IteratorAggregate
+{
+    private $datePeriod;
+
+    /**
+     * Creates a new DateTime period.
+     *
+     * @param \DateTime $startDate
+     * @param \DateTime $endDate
+     */
+    public function __construct(\DateTime $startDate, \DateTime $endDate)
+    {
+        // We need to add one day to include the end date as well
+        $endDate = $endDate->modify('+1 day');
```

Comment:
$endDate needs to be cloned otherwise the constructor will modify it (it's not immutable)